From 7d287a5d5f809ee34fd510bec8beeb1bb1eb07cb Mon Sep 17 00:00:00 2001
From: ShuFan Lee <shufan_lee@richtek.com>
Date: Mon, 12 Mar 2018 19:55:24 +0800
Subject: [PATCH] staging: typec: modify parameter of tcpci_irq
Patch-mainline: v4.17
Git-commit: 7d287a5d5f809ee34fd510bec8beeb1bb1eb07cb
References: FATE#326325

The parameter, dev_id, of tcpci_irq should be tcpci_chip.
Remove definition of TCPC_CC_STATUS_DRPRST in tcpci.h.

Fixes: 8f9439022648("staging: typec: handle vendor defined part and modify drp toggling flow")
Signed-off-by: ShuFan Lee <shufan_lee@richtek.com>
Reviewed-by: Li Jun <jun.li@nxp.com>
Tested-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/staging/typec/tcpci.c | 4 ++--
 drivers/staging/typec/tcpci.h | 1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index 804374056a1e..076d97eaff6f 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -451,9 +451,9 @@ EXPORT_SYMBOL_GPL(tcpci_irq);
 
 static irqreturn_t _tcpci_irq(int irq, void *dev_id)
 {
-	struct tcpci *tcpci = dev_id;
+	struct tcpci_chip *chip = dev_id;
 
-	return tcpci_irq(tcpci);
+	return tcpci_irq(chip->tcpci);
 }
 
 static const struct regmap_config tcpci_regmap_config = {
diff --git a/drivers/staging/typec/tcpci.h b/drivers/staging/typec/tcpci.h
index a2c1754ca222..34c865f0dcf6 100644
--- a/drivers/staging/typec/tcpci.h
+++ b/drivers/staging/typec/tcpci.h
@@ -59,7 +59,6 @@
 #define TCPC_POWER_CTRL_VCONN_ENABLE	BIT(0)
 
 #define TCPC_CC_STATUS			0x1d
-#define TCPC_CC_STATUS_DRPRST		BIT(5)
 #define TCPC_CC_STATUS_TERM		BIT(4)
 #define TCPC_CC_STATUS_CC2_SHIFT	2
 #define TCPC_CC_STATUS_CC2_MASK		0x3
-- 
2.16.4

